job_executor/model/metadata_all.py

```diff
@@ -28,26 +28,32 @@ class MetadataAll(CamelModel):
     def __iter__(self):
         return iter(
             [
-                Metadata(**data_structure.dict(by_alias=True))
+                Metadata(
+                    **data_structure.model_dump(
+                        by_alias=True, exclude_none=True
+                    )
+                )
                 for data_structure in self.data_structures
             ]
         )
 
     def get(self, dataset_name: str) -> Union[Metadata, None]:
         for metadata in self.data_structures:
             if metadata.name == dataset_name:
-                return Metadata(**metadata.dict())
+                return Metadata(**metadata.model_dump())
```

Comment:
Let's use the aliases, and exclude nones here too for the sake of consistency 🙏🏻